### PR TITLE
Support FrameworkReference items

### PR DESCRIPTION
**Pull request by dsplaisted**
Add target which transforms `FrameworkReference` items into implicit NuGet package references (for the corresponding targeting packs) as well as entries in the runtimeconfig.json.

Multiple shared frameworks are supported per the design for .NET Core 3.0 specified [here](https://github.com/steveharter/core-setup/blob/master/Documentation/design-docs/multilevel-sharedfx-lookup.md).

@livarcocc @peterhuene @steveharter for review

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenADependencyContextBuilder.cs [hunk 1]
```diff
@@ -51,6 +51,7 @@ public class GivenADependencyContextBuilder
                 FrameworkConstants.CommonFrameworks.NetCoreApp10,
                 runtime,
                 Constants.DefaultPlatformLibrary,
+                null,
```
Comment (Pilchie) [inline src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenADependencyContextBuilder.cs]: Nit: suggest using a named param for the literal as for `isSelfContained:`

**Comment by dsplaisted** [comment]
@dotnet-bot test this please

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            public string DefaultRuntimeFrameworkVersion => _item.GetMetadata("DefaultRuntimeFrameworkVersion");
+            public string LatestRuntimeFrameworkVersion => _item.GetMetadata("LatestRuntimeFrameworkVersion");
+
+            //  The ID of the targeting pack NuGet package to reference
+            public string TargetingPackName => _item.GetMetadata("TargetingPackName");
+            public string TargetingPackVersion => _item.GetMetadata("TargetingPackVersion");
+        }
+    }
+
+    public class ReportUnknownFrameworkReferences : TaskBase
```
Comment (livarcocc) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs]: Why not put this in its own file?

**Review by livarcocc — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs [hunk 3]
```diff
@@ -95,19 +98,50 @@ private void WriteRuntimeConfig(ProjectContext projectContext)
...
-        private void AddFramework(RuntimeOptions runtimeOptions, ProjectContext projectContext)
+        private void AddFrameworks(RuntimeOptions runtimeOptions, ProjectContext projectContext)
         {
             if (projectContext.IsFrameworkDependent)
             {
-                var platformLibrary = projectContext.PlatformLibrary;
-                if (platformLibrary != null)
+                runtimeOptions.tfm = TargetFramework;
+
+                if (projectContext.RuntimeFrameworks == null)
```
Comment (peterhuene) [inline src/Tasks/Microsoft.NET.Build.Tasks/GenerateRuntimeConfigurationFiles.cs]: Can the array be zero-length?

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                    unresolvedFrameworkReferences.Add(frameworkReference.ItemSpec);
+                }
+            }
+
+            if (packageReferencesToAdd.Any())
+            {
+                PackageReferencesToAdd = packageReferencesToAdd.ToArray();
+            }
+
+            if (runtimeFrameworks.Any())
```
Comment (peterhuene) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs]: This appears to answer my question above: no (at least not from the output of this task).  We still may want to guard against a non-null, zero-length input for the `GenerateRuntimeConfigurationFiles` task.

**Review by peterhuene — APPROVED** [review]
:+1: Just the one question regarding handling of non-null, zero-length `RuntimeFrameworks` array in `GenerateRuntimeConfigurationFiles`.

**Comment by dsplaisted** [comment]
@dotnet-bot test Windows_NT Release

**Pull request merged**